zipline/data/bundles/core.py

```diff
@@ -586,8 +586,8 @@ def clean(name,
             def should_clean(name):
                 dt = from_bundle_ingest_dirname(name)
                 return (
-                    (before is not None and dt < before) or
-                    (after is not None and dt > after)
+                    (before is None or dt < before) and
+                    (after is None or dt > after)
```

Comment:
why invert all of these conditions?

Reply:
This was my attempt to fix the noop `clean` commands (when I passed no args). I'll instead raise `BadClean` in that case.